File: shoring/flywheel/src/main/java/com/panopset/flywheel/TemplateFile.java
package com.panopset.flywheel;

import java.io.File;
import com.panopset.compat.Fileop;

public final class TemplateFile extends TemplateArray {
  private final String fn;

  public TemplateFile(final File file) {
    super(Fileop.readLines(file));
    fn = Fileop.getCanonicalPath(file);
  }

  @Override
  public String getName() {
    return fn;
  }
}

File: code/shoring/flywheel/src/main/kotlin/com/panopset/flywheel/TemplateFile.kt
package com.panopset.flywheel

import com.panopset.compat.Fileop
import java.io.File

class TemplateFile(file: File) : TemplateArray(Fileop.readLines( file)) {
    override val name: String = Fileop.getCanonicalPath(file)
}
